src/lib/telemetry/TelemetryReporter.ts

```diff
@@ -69,6 +70,7 @@ export class TelemetryReporter {
     try {
       const telemetryObj = await readJson(telemetryObjFilePath)
       await this.reportErrors(telemetryObj.errors)
+      await this.reportMetrics(telemetryObj.metrics)
```

Comment:
What if errors are reported successfully but metrics aren't? Maybe we should move the error logic to `reportErrors` and `reportMetrics`?
``` ts
public async reportMetrics(metrics: MetricReport[]) {
  try {
    await this.telemetryClient.reportMetrics(metrics)
  } catch(err) {
    await this.dataPendingLock.lock()
    await writeJson(join(TelemetryReporter.PENDING_DATA_DIR, `metricsRemaining-randomBytes(8).toString('hex')}`))
    await this.createTelemetryReporterMetaError(err)
    await this.dataPendingLock.unlock()
  }
}
```